### PR TITLE
Add API to delete alerts within a context and update 'alerts' view API to allow filtering by context

### PR DESCRIPTION
**Pull request by kingthorin**
- AlertAPI.java > Update "alerts" view to optionally accept and filter based on a context name. Add an alerts action "deleteAlerts" which accepts a context name and deletes any alerts for which the URL is in context (optionally filtered by baseurl or risk).
- Messages.properties > Add the new key value pairs for the API and param descriptions. Also define param.baseurl for other API endpoints which leverage baseurl.

Fixes zaproxy/zaproxy#1926

<details>
<summary>Images of original impl</summary>

image
image

</details>

### CODE REVIEW

**Comment by psiinon** [comment]
How about a deleteAlerts endpoint instead, with the same params as viewAlerts?
That would still solve the issue but would be more flexible?

**Comment by kingthorin** [comment]
I'm not sure how we'd handle pagination on deletions.

**Comment by kingthorin** [comment]
Okay, I'll look at rejigging that.

**Comment by kingthorin** [comment]
Do we really need baseUrl when we're using contexts? Which should take precedence?

**Comment by kingthorin** [comment]
Or deleteAlerts (with baseUrl and Risk) and deleteAlertsForContext (with context and risk)?

**Comment by psiinon** [comment]
I now prefer fewer API endpoints using params more :)
Others may disagree of course!
I think both baseUri AND context should apply if included.
A context can have lots of URL and so a baseUri can help be more specific.
And URLs can be excluded from contexts so a context can be more specific that a baseUri.

**Comment by kingthorin** [comment]
Tweaked

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java [hunk 1]
```diff
@@ -61,6 +62,7 @@ public class AlertAPI extends ApiImplementor {
     public static final String PREFIX = "alert";
 
     private static final String ACTION_DELETE_ALL_ALERTS = "deleteAllAlerts";
+    private static final String ACTION_DELETE_ALERTS_FOR_CONTEXT = "deleteAlertsForContext";
```
[the remark this answers is not in the document]
Comment (psiinon) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:65]: And is optional 😁

**Comment by kingthorin** [comment]
Tweaked the endpoint name. How does this sound for baseUrl (throughout the API)?

> A partial or full URL used to filter based on a "starts with" behaviour.

(Or "check" or "functionality" instead of "behaviour")

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kingthorin) [reply zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java]: Tweaked the text too.

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java [hunk 2]
```diff
@@ -61,6 +62,7 @@ public class AlertAPI extends ApiImplementor {
     public static final String PREFIX = "alert";
 
     private static final String ACTION_DELETE_ALL_ALERTS = "deleteAllAlerts";
+    private static final String ACTION_DELETE_ALERTS_FOR_CONTEXT = "deleteAlerts";
```
Comment (psiinon) [inline zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:65]: `ACTION_DELETE_ALERTS` 😁

File: zap/src/main/resources/org/zaproxy/zap/resources/Messages.properties [hunk 3]
```diff
@@ -32,6 +32,10 @@ alert.api.action.addAlert.param.solution = The solution for the alert.
 alert.api.action.addAlert.param.wascId = The WASC identifier associated with the alert.
 alert.api.action.deleteAlert = Deletes the alert with the given ID. 
 alert.api.action.deleteAlert.param.id = 
+alert.api.action.deleteAlerts = Deletes all the alerts optionally filtered by URL which fall with in the Context with the provided name, risk, or base URL.
+alert.api.action.deleteAlerts.param.baseurl =
```
Comment (psiinon) [inline zap/src/main/resources/org/zaproxy/zap/resources/Messages.properties:36]: `The highest URL in the Sites tree under which alerts should be deleted.` ?

**Comment by kingthorin** [comment]
Updated.

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**